go/arrow/array/booleanbuilder_test.go

```diff
@@ -43,3 +43,54 @@ func TestBooleanBuilder_AppendValues(t *testing.T) {
 	assert.Equal(t, exp, got)
 	a.Release()
 }
+
+func TestBooleanBuilder_Empty(t *testing.T) {
+	mem := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer mem.AssertSize(t, 0)
+
+	ab := array.NewBooleanBuilder(mem)
+	defer ab.Release()
+
+	want := tools.Bools(1, 1, 0, 1, 1, 0, 1, 0)
+
+	ab.AppendValues(want, nil)
+
+	boolValues := func(a *array.Boolean) []bool {
+		vs := make([]bool, a.Len())
+		for i := range vs {
+			vs[i] = a.Value(i)
+		}
+		return vs
+	}
+
+	a := ab.NewBooleanArray()
+	ab.Release()
+	assert.Equal(t, want, boolValues(a))
+	a.Release()
+
+	a = ab.NewBooleanArray()
+	assert.Zero(t, a.Len())
+	a.Release()
+
+	ab.AppendValues(nil, nil)
+	a = ab.NewBooleanArray()
+	assert.Zero(t, a.Len())
+	a.Release()
+
+	ab.AppendValues([]bool{}, nil)
```

Comment:
I would maybe move the `AppendValues` with empty elements to the top of the test, since I think the panic was due to not initializing the null bitmap. and that is not tested if we initialize it with `AppendValues(want, ...)`. Although maybe `NewBooleanArray` resets these buffers so it doesn't matter?

Reply:
(another) good point.
fixed.